torch/distributed/_composable/fsdp/_fsdp_collectives.py

```diff
@@ -336,6 +336,11 @@ def foreach_reduce(
                     new_sharded_grad
                 )
                 fsdp_param.sharded_param.grad = new_sharded_dtensor_grad
+            for hook in (
+                getattr(fsdp_param.sharded_param, "_post_accumulate_grad_hooks", {})
+                or {}
+            ).values():
```

Comment:
@yf225 What should we do here? Dynamo does not support this `getattr` call.

```
File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 1480, in CALL_FUNCTION
2024-06-25T20:11:10.8946722Z     self.call_function(fn, args, {})
2024-06-25T20:11:10.8947571Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 754, in call_function
2024-06-25T20:11:10.8947838Z     self.push(fn.call_function(self, args, kwargs))
2024-06-25T20:11:10.8948692Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/variables/builtin.py", line 963, in call_function
2024-06-25T20:11:10.8948882Z     return handler(tx, args, kwargs)
2024-06-25T20:11:10.8949707Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/variables/builtin.py", line 712, in <lambda>
2024-06-25T20:11:10.8949968Z     return lambda tx, args, kwargs: obj.call_function(
2024-06-25T20:11:10.8950846Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/variables/builtin.py", line 963, in call_function
2024-06-25T20:11:10.8951019Z     return handler(tx, args, kwargs)
2024-06-25T20:11:10.8951892Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/variables/builtin.py", line 847, in builtin_dipatch
2024-06-25T20:11:10.8952082Z     rv = fn(tx, args, kwargs)
2024-06-25T20:11:10.8952978Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/variables/builtin.py", line 765, in call_self_handler
2024-06-25T20:11:10.8953209Z     result = self_handler(tx, *args, **kwargs)
2024-06-25T20:11:10.8954068Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/variables/builtin.py", line 1564, in call_getattr
2024-06-25T20:11:10.8954323Z     hasattr_var = self.call_hasattr(tx, obj, name_var)
2024-06-25T20:11:10.8955350Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/variables/builtin.py", line 1471, in call_hasattr
2024-06-25T20:11:10.8955649Z     return obj.call_hasattr(tx, name)
2024-06-25T20:11:10.8956490Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/variables/base.py", line 296, in call_hasattr
2024-06-25T20:11:10.8956802Z     unimplemented(f"hasattr {self.__class__.__name__} {name}")
2024-06-25T20:11:10.8957567Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/exc.py", line 221, in unimplemented
2024-06-25T20:11:10.8957753Z     raise Unsupported(msg)
2024-06-25T20:11:10.8958218Z torch._dynamo.exc.Unsupported: hasattr TensorVariable _post_accumulate_grad_hooks
```

Reply:
@awgu I believe there is a way to make this work - for now would you like to skip this path under compile? I can come back to make this work for compile later

Reply:
Yep skipping this for now is good.